domain/sysvar_cache.go

```diff
@@ -63,36 +59,36 @@ func (svc *SysVarCache) rebuildCacheIfNeeded(ctx sessionctx.Context) (err error)
 // GetSessionCache gets a copy of the session sysvar cache.
 // The intention is to copy it directly to the systems[] map
 // on creating a new session.
-func (svc *SysVarCache) GetSessionCache(ctx sessionctx.Context) (map[string]string, error) {
-	if err := svc.rebuildCacheIfNeeded(ctx); err != nil {
+func (do *Domain) GetSessionCache() (map[string]string, error) {
+	if err := do.rebuildSysVarCacheIfNeeded(); err != nil {
 		return nil, err
 	}
-	svc.RLock()
-	defer svc.RUnlock()
+	do.sysVarCache.RLock()
+	defer do.sysVarCache.RUnlock()
 	// Perform a deep copy since this will be assigned directly to the session
-	newMap := make(map[string]string, len(svc.session))
-	for k, v := range svc.session {
+	newMap := make(map[string]string, len(do.sysVarCache.session))
+	for k, v := range do.sysVarCache.session {
 		newMap[k] = v
 	}
 	return newMap, nil
 }
 
 // GetGlobalVar gets an individual global var from the sysvar cache.
-func (svc *SysVarCache) GetGlobalVar(ctx sessionctx.Context, name string) (string, error) {
-	if err := svc.rebuildCacheIfNeeded(ctx); err != nil {
+func (do *Domain) GetGlobalVar(name string) (string, error) {
```

Comment:
I think we can do a much smaller change by replacing `SysVarCache` with `sysVarCache`, instead of directly to `Domain`.

And embedding this struct will be able to `do.GetGlobalVar` directly.

Reply:
See the comment in the description:

> This required a slight code rework to prevent cyclic dependencies. The sysvar cache funcs are now part of the domain struct and not the SysVarCache struct.

Getting a session from the pool, requires `domain`, which I can either add as a field to `sysVarCache` (seems unneccesary) or get with `dom, err := session.GetDomain(nil)` (causes cyclic dependencies)

Reply:
Makes sense. In fact, I am thinking about passing from the outside. However it will need a lot `sysSessionPool.Get()`.